components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py

```diff
@@ -0,0 +1,168 @@
+            request["MaxConcurrentTransforms"] = inputs.max_concurrent
+
+        if inputs.max_payload or inputs.max_payload == 0:
+            request["MaxPayloadInMB"] = inputs.max_payload
+
+        if inputs.batch_strategy:
+            request["BatchStrategy"] = inputs.batch_strategy
+        else:
+            request.pop("BatchStrategy")
+
+        request["Environment"] = inputs.environment
+
+        if inputs.data_type:
+            request["TransformInput"]["DataSource"]["S3DataSource"][
+                "S3DataType"
+            ] = inputs.data_type
+
+        request["TransformInput"]["DataSource"]["S3DataSource"][
+            "S3Uri"
+        ] = inputs.input_location
+        request["TransformInput"]["ContentType"] = inputs.content_type
+
+        if inputs.compression_type:
+            request["TransformInput"]["CompressionType"] = inputs.compression_type
+
+        if inputs.split_type:
+            request["TransformInput"]["SplitType"] = inputs.split_type
+
+        request["TransformOutput"]["S3OutputPath"] = inputs.output_location
+        request["TransformOutput"]["Accept"] = inputs.accept
+        request["TransformOutput"]["KmsKeyId"] = inputs.output_encryption_key
+
+        if inputs.assemble_with:
+            request["TransformOutput"]["AssembleWith"] = inputs.assemble_with
+        else:
+            request["TransformOutput"].pop("AssembleWith")
+
+        request["TransformResources"]["InstanceType"] = inputs.instance_type
+        request["TransformResources"]["InstanceCount"] = inputs.instance_count
+        request["TransformResources"]["VolumeKmsKeyId"] = inputs.resource_encryption_key
+        request["DataProcessing"]["InputFilter"] = inputs.input_filter
+        request["DataProcessing"]["OutputFilter"] = inputs.output_filter
+
+        if inputs.join_source:
+            request["DataProcessing"]["JoinSource"] = inputs.join_source
+
+        self._enable_tag_support(request, inputs)
+
+        return request
+
+    def _submit_job_request(self, request: Dict) -> object:
+        return self._sm_client.create_transform_job(**request)
+
+    def _after_submit_job_request(
+        self,
+        job: object,
+        request: Dict,
+        inputs: SageMakerTransformInputs,
+        outputs: SageMakerTransformOutputs,
+    ):
+        logging.info(f"Created Transform Job with name: {self._transform_job_name}")
+        logging.info(
+            "Transform job in SageMaker: https://{}.console.aws.amazon.com/sagemaker/home?region={}#/jobs/{}".format(
+                inputs.region, inputs.region, self._transform_job_name,
+            )
+        )
+        logging.info(
+            "CloudWatch logs: https://{}.console.aws.amazon.com/cloudwatch/home?region={}#logStream:group=/aws/sagemaker/TransformJobs;prefix={};streamFilter=typeLogStreamPrefix".format(
+                inputs.region, inputs.region, self._transform_job_name,
+            )
+        )
+
+
+if __name__ == "__main__":
+    import sys
```

Comment:
1. more import to top of file
2. Can we avoid using sys.argv ?

Reply:
Import inside `__main__` is common because if we want to use this file as a module, we don't want to import that library. But I'm not hard set on that and happy to move it.

I did a bit of Googling and haven't found any other way, at least nothing documented, about passing command line arguments to `argparse`. Is there any good reason to not use `sys.argv` though?